lib/modules/manager/gomod/artifacts.ts

```diff
@@ -410,3 +413,19 @@ export async function updateArtifacts({
     ];
   }
 }
+
+async function getGoConstraints(
+  goModFileName: string
+): Promise<string | undefined> {
+  let content = (await readLocalFile(goModFileName, 'utf8')) ?? null;
+  if (!content) {
+    return undefined;
+  }
+  content = content.replace(
+    regEx(/(?<goText>go\s*)(?<gover>\d+.\d+)/),
+    '$<goText>^$<gover>'
+  );
+  const re = regEx(/go\s*(?<gover>\^\d+.\d+)/);
+  const match = re.exec(content);
+  return match?.groups?.gover;
```

Comment:
Avoid using two regexes by first extracting the version/value from the file, and then prepend `^`.